File: solutions_1-10/Euler_2.py
## sum of even fibonacci numbers below 4 million ? ##

import timeit


start = timeit.default_timer()

def Euler_2(limit):
    a1 = 1
    b1 = 1
    b2 = 0
    while b2 <= limit:
        b2 = a1 + b1
        a1 = b1
        b1 = b2
        if b2 % 2 == 0:
            yield b2

print "Answer: %s" % sum([i for i in Euler_2(4000000)])
stop = timeit.default_timer()
print "Time: %s" % str(stop - start)

File: Java/java_solutions_1-10/Euler_2.java
/*
/   Project Euler problem #2 Even Fibonacci numbers
/
/   Find the sum of the even fibonacci terms <= 4 million
/
*/
import java.util.stream.IntStream;
import java.util.OptionalInt;

public class Euler_2
{
    public static long fibonacci(long fnum, long idx, int limit, long total) {
        if (fnum >= limit) {
            return total;
        }

        if (fnum % 2 == 0) {
            total += fnum;
        }

        return fibonacci(fnum + idx, fnum, limit, total);
    }

    public static void main(String[] args)
    {
        long start = System.nanoTime();
        long sumFibonacci = fibonacci(3, 2, 4000000, 0);
        long stop = System.nanoTime();
        System.out.println("Answer: " + sumFibonacci);
        System.out.printf("Time: %.7f\n", ((float) stop - start));
    }
}
            
